test/Raven/Tests/ReprSerializerTest.php

```diff
@@ -109,4 +109,45 @@ public function testSerializeRoundedFloat()
         $this->assertInternalType('string', $result);
         $this->assertEquals('12345.0', $result);
     }
+
+    /**
+     * @dataProvider serializableDataProvider
+     */
+    public function testSerializeSerializableObject($expected)
+    {
+        $serializer = new Raven_ReprSerializer();
+
+        $mock = $this->getMockBuilder('Raven_Serializable')
+                     ->getMock();
+        $mock->method('toSentry')->willReturn($expected);
+
+        $result = $serializer->serialize($mock);
+        $this->assertEquals($expected, $result);
+    }
+
+    /**
+     * @dataProvider serializableDataProvider
+     */
+    public function testGenericSerializableObject($expected)
+    {
+        $serializer = new Raven_ReprSerializer();
+
+        $mock = $this->getMockBuilder('stdClass')
+                     ->setMethods(array('toDebugContext'))
+                     ->getMock();
+        $mock->method('toDebugContext')->willReturn($expected);
+
+        $result = $serializer->serialize($mock);
+        $this->assertEquals($expected, $result);
```

Comment:
i dont entirely remember the logic behind how i wrote serializers, but here were the goals:

1. certain things should serialize representations of values (e.g. they "look" like the real value). This is stack args and extra context
2. Basic serialization should coerce to simple types. That should exist on our core JSON payloads, tags, etc.